homeassistant/components/template/binary_sensor.py

```diff
@@ -321,7 +322,7 @@ def is_on(self) -> bool | None:
         """Return true if sensor is on."""
         return self._state
 
-    @property
+    @cached_property
     def device_class(self) -> BinarySensorDeviceClass | None:
```

Comment:
It looks like this can be set in the constructor in a separate PR

Reply:
#100301